calico/felix/futils.py

```diff
@@ -79,12 +80,29 @@ def __init__(self, message, args, retcode, stdout, stderr, input=None):
         self.input = input
 
     def __str__(self):
+        input = self.input
```

Comment:
It can't cause problems here (I think), but I always get a little twitchy whenever I see variables with the same name as the Python builtins.

I'm also confused &ndash; where are we actually using this variable? What's the difference between this and `safe_truncate()`?